src/frontend/Ast_to_Mir.ml

```diff
@@ -136,14 +138,48 @@ let trans_printables mloc (ps : Ast.typed_expression Ast.printable list) =
       | Ast.PExpr e -> trans_expr e)
     ps
 
+let unsizedtype_of_indexed ut
```

Comment:
What motivated not using the Semantic check function anymore?

Reply:
During the refactor, the function in `Semantic_check` was changed to return an `unsizedtype Validate.t` since there is a possibility that the indexed type is not semantically valid. 

In`Ast_to_Mir` we are only ever working with semantically correct ASTs so we should always be able to return an `unsizedtype`. Any deviation from this is genuinely exceptional behavior so we should use exceptions here to handle that case.

Whilst I could have reused the function inside `Semantic_check`, converted the `Validate.t` to a result and then raised an exception in the `Error` case, it would have meant exposing the `Validate` module outside of the semantic check module and violating DRY seemed a little nicer.

An alternative would be to expose a function `unsizedtype_of_indexed_exn` from `Semantic_check` although it would be a little odd in terms of the minimal API that module exposes. 

Happy to do the latter if you think it makes more sense?

Reply:
Is there somewhere neutral that we can put it so that both files have access to it? It seems unnecessary to duplicate that function, particularly because I remember that its logic is a bit subtle. This would be my only real requested change before this can be merged.

Reply:
Sure - I think the second option above is most sensible i.e. expose expose a function `unsizedtype_of_indexed_exn` which unsafely coverts the `unsizedtype Validate.t` into an `unsizedtype`.